packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -47,6 +55,15 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
             }
         });
 
+        commands.registerCommand(VscodeCommands.DIFF, {
+            isVisible: () => true,
+            // tslint:disable-next-line: no-any
+            execute: async uris => {
+                const [left, right] = uris;
+                await this.diffService.openDiffEditor(left, right);
```

Comment:
@vzhukovskii, why not just executing `WorkspaceCommands.FILE_COMPARE` here? cf. `VscodeCommands.OPEN` definition. I think that's what was mentioned by @kittaakos.

Reply:
@AlexTugarev I'm not sure if it is the correct handling of vscode's command. As I understand, handler of `WorkspaceCommands.FILE_COMPARE` wraps incoming handler and uses selection object to extract uris and pass them into the last one, so we can not pass own uris into this command. On the other hand we should have an ability not to become attached to global selection and use incoming uris from the arguments. Like here: [definition](https://github.com/Microsoft/vscode/blob/5fb0f1ead7c411dd678bcd96df70f452ba73c31b/src/vs/workbench/api/node/apiCommands.ts#L89) and [sample of usage](https://github.com/Microsoft/vscode/blob/dcb3c339c8d02aa8b218c3e8c89d313491f6cc2c/extensions/git/src/commands.ts#L289). So I thought, that it would be better to extract the code which opens actual editor into dedicate class and call it from two places, the first one which uses selection and the second one which takes incoming uris as an argument. WDYT?